modular_skyrat/modules/delam_emergency_stop/code/scram.dm

```diff
@@ -0,0 +1,435 @@
+/datum/controller/subsystem/nightshift/proc/suck_light_power()
+	SSnightshift.can_fire = FALSE
+	for(var/obj/machinery/power/apc/light_to_suck as anything in SSmachines.get_machines_by_type(/obj/machinery/power/apc))
+		light_to_suck.cell.charge -= DELAM_MACHINE_POWER_CONSUMPTION
+		light_to_suck.lighting = APC_CHANNEL_OFF
+		light_to_suck.nightshift_lights = TRUE
+		light_to_suck.update_appearance()
+		light_to_suck.update()
+
+/// Restores the lighting after the delam suppression
+/datum/controller/subsystem/nightshift/proc/restore_light_power()
+	for(var/obj/machinery/power/apc/light_to_restore as anything in SSmachines.get_machines_by_type(/obj/machinery/power/apc))
+		light_to_restore.lighting = APC_CHANNEL_AUTO_ON
+		light_to_restore.update_appearance()
+		light_to_restore.update()
+
+/// A big red button you can smash to stop the supermatter engine, oh how tempting!
+/obj/machinery/button/delam_scram
+	name = "supermatter emergency stop"
+	desc = "Your last hope to try and save the crystal during a delamination.<br>\
+	While it is indeed a big red button, pressing it outside of an emergency \
+	will probably get the engineering department out for your blood."
+	icon = 'modular_skyrat/modules/delam_emergency_stop/icons/scram.dmi'
+	can_alter_skin = FALSE
+	silicon_access_disabled = TRUE
+	resistance_flags = FREEZE_PROOF | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF
+	light_color = LIGHT_COLOR_INTENSE_RED
+	light_power = 0.7
+	///one use only!
+	var/button_stage = BUTTON_IDLE
+	///our internal radio
+	var/obj/item/radio/radio
+	///radio key
+	var/radio_key = /obj/item/encryptionkey/headset_eng
+
+/obj/machinery/button/delam_scram/Initialize(mapload)
+	. = ..()
+	radio = new(src)
+	radio.keyslot = new radio_key
+	radio.set_listening(FALSE)
+	radio.recalculateChannels()
+
+/obj/machinery/button/delam_scram/Destroy()
+	QDEL_NULL(radio)
+	return ..()
+
+/obj/machinery/button/delam_scram/screwdriver_act(mob/living/user, obj/item/tool)
+	return TRUE
+
+/obj/machinery/button/delam_scram/emag_act(mob/user)
+	return
+
+/// Proc for arming the red button, it hasn't been pushed yet
+/obj/machinery/button/delam_scram/attack_hand(mob/user, list/modifiers)
+	. = ..()
+	if(.)
+		return
+
+	// Give them a cheeky instructions card. But only one! If you lost it, question your engineering prowess in this moment
+	if(button_stage == BUTTON_IDLE)
+		visible_message(span_danger("A biscuit card falls out of the [src]!"))
+		user.put_in_hands(new /obj/item/folder/biscuit/confidential/delam(get_turf(user)))
+		button_stage = BUTTON_AWAKE
+		return
+
+	if(button_stage != BUTTON_AWAKE)
+		return
+
+	// For roundstart only, after that it's on you!
+	if(world.time - SSticker.round_start_time > 30 MINUTES)
+		playsound(src.loc, 'sound/misc/compiler-failure.ogg', 50, FALSE, 15)
+		audible_message(span_danger("The [src] makes a series of sad beeps. The internal charge only lasts about 30 minutes... what a feat of engineering!"))
```

Comment:
Isn't this a copy-paste of the message that was there higher up in this file?
If so, do we need the double validation?

Reply:
It needs a check here so that we don't annoy people with the radio announcement. It probably doesn't need the second check, it was protecting against a theoretical edge case where someone manages to send the signal somehow even when the button is broken.